logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -293,7 +293,18 @@ private boolean executeStoragePolicy(int eventPayloadSize) {
         if (!exceedMaxQueueSize(eventPayloadSize)) {
             return false;
         }
-        
+
+        // load currentQueueSize from filesystem because there could be a consumer
+        // that's already cleaning
+        try {
+            this.currentQueueSize.set(computeQueueSize());
+            if (!exceedMaxQueueSize(eventPayloadSize)) {
+                return false;
+            }
+        } catch (IOException ex) {
+            logger.error("Can't read DLQ size from filesystem", ex);
```

Comment:
Maybe a warn here, and an indication of what the implications of not being able to read the DLQ size are

Also, I think we should return `false` here - otherwise a failure to read DLQ size will cause a message or segment to be dropped

```suggestion
            logger.warn("Unable to determine DLQ size, skipping storage policy check", ex);
            return false;
```

Reply:
Good point, maybe better to be conservative and exceed the configured max size. If the machine has permanent IO problem, then it would report errors also somewhere  else.

Reply:
Maybe the 
```java
if (!exceedMaxQueueSize(eventPayloadSize)) {
    return false;
}
```
could be extracted out of the `try .. catch` to improve readability, or express better the intention.